### ISSUE TITLE
How is aurelia.setRoot call working without using Platform.moduleName?

### ISSUE THREAD

**Issue by AshleyGrant** [greensponge/aurelia-dotnet-template#2]
https://github.com/greensponge/aurelia-dotnet-template/blob/edc896910b1354dc190b267f3c998ee213502031/src/boot.ts#L24

Do you know why this line works without needing to call `Platform.moduleName`? I've been led to believe that we have to use `Platform.moduleName` to allow Webpack to do it's static analysis, so I'm curious how this is working without that call?

**Comment by greensponge** [comment]
Excellent question, I wasn't 100% sure of this myself, so I compared my project to a new au-cli default TypeScript project where `setRoot()` fails if you don't use `PLATFORM.moduleName` and narrowed it down to this line in `webpack.config.js`: 

`new GlobDependenciesPlugin({ "boot": ["src/**/*.{ts,html}"] })`

Similarly if you were to use `new GlobDependenciesPlugin({ "main": ["src/**/*.{ts,html}"] })` in a new au-cli project it would also work without wrapping the contents of `setRoot()` with `PLATFORM.moduleName`.

I'm sure @jods4 could give you a better answer, but here is his documentation of this plugin which is pretty much all I know about it:

>new GlobDependenciesPlugin({ [module: string]: string | string[] })
This module is similar to ModuleDependenciesPlugin above, but instead of adding specific dependencies to a module, it adds every file that matches one (or several) Globs. Note that globs are evaluated relative to your webpack config location.
This is what AureliaPlugin uses to implement its includeAll option.
[source](https://github.com/aurelia/webpack-plugin/wiki/Secondary-plugins)

**Comment by jods4** [comment]
That is right, `GlobDependenciesPlugin` is a way to create Aurelia dependencies _en masse._

It is primarily meant for small projects, demos, proof of concepts -- or ease migration of large existing projects.

Ideally, you would not rely on it for new projects, as it prevents code splitting and disables several optimizations on every file it matches, even those not imported by Aurelia.

As far as startup goes, look at this note:
https://github.com/aurelia/webpack-plugin/wiki/Webpack-config#1-entry-point

TL;DR: there is an option to account for convention and if you use `setRoot` explicitly you should use `moduleName`.

**Comment by greensponge** [comment]
Thanks for taking the time to reply @jods4 ! This is really good information, I will change this template accordingly so I don't unintentionally cripple myself or anyone else from the start.

**Comment by veikkoeeva** [comment]
Should this be closed now? :)

**Comment by AshleyGrant** [comment]
Yep
